x-pack/plugins/osquery/server/routes/status/create_status_route.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OSQUERY_INTEGRATION_NAME } from '../../../common';
+import { IRouter } from '../../../../../../src/core/server';
+import { OsqueryAppContext } from '../../lib/osquery_app_context_services';
+
+export const createStatusRoute = (router: IRouter, osqueryContext: OsqueryAppContext) => {
+  router.get(
+    {
+      path: '/internal/osquery/status',
+      validate: false,
+    },
+    async (context, request, response) => {
+      const soClient = context.core.savedObjects.client;
+      const isSuperUser = osqueryContext.security.authc
+        .getCurrentUser(request)
+        ?.roles.includes('superuser');
```

Comment:
I am wondering if we should expose a method in Fleet to do that something like `isAuthorized(request)`. I think in the future we want to allow non superuser with enough permissions.

Reply:
that would be helpful @nchaulet. We are also working on allowing to use Osquery for non superuser